### PR TITLE
show warning and retry signing process in case of errors

### PR DESCRIPTION
**Pull request by ChristophWurst**
We could also try to diagnose the problem with the help of error codes from https://developers.yubico.com/U2F/Libraries/Client_error_codes.html

### CODE REVIEW

File: templates/challenge.php [hunk 1]
```diff
@@ -13,5 +13,11 @@
...
 <img src="<?php print_unescaped(image_path('twofactor_u2f', 'app.svg')); ?>">
-<p><?php p($l->t('Please plug in your U2F device and press the device button to authorize.')) ?></p>
+<p id="u2f-info">
+	<?php p($l->t('Please plug in your U2F device and press the device button to authorize.')) ?>
+</p>
+<p id="u2f-error"
+   style="display: none">
+	<strong><?php p($l->t('An error occurred. Please try again.')) ?></strong>
+</p>
 <p><em><?php p($l->t('Chrome is the only browser that supports U2F devices. You need to install the "U2F Support Add-on" on Firefox to use U2F.')) ?></em></p>
```
Comment (My1) [inline templates/challenge.php]: Opera by now also does U2F. I dunno which versions but I know it works, I use it myself.
but nice that you are referencing that FF has a U2F Addon.

Comment (ChristophWurst) [inline templates/challenge.php]: Right, good point. However, I'm not a big fan of opera and apparently only about one percent of internet users is using opera. Not sure if it's worth to mention here as it has also not been tested on a opera browser yet.

**Comment by ChristophWurst** [comment]
Simplest test: don't do anything on the 2FA challenge page. The U2F sign request will time out (error 5 https://developers.yubico.com/U2F/Libraries/Client_error_codes.html). An error message should be shown and hidden after 5sec again. U2F login should then work.

File: templates/challenge.php (hunk 1 above)
Comment (My1) [inline templates/challenge.php]: well I am running opera daily and I can help with opera testing. right now I have it running on opera beta 43 but I have a stable at home which I can also try.

Comment (ChristophWurst) [inline templates/challenge.php]: Awesome! I'll update the description in another PR

Comment (My1) [inline templates/challenge.php]: also if you need help with testing just ask, I have 3 U2F sticks at hand (yubikey 4 nano and 2 very cheap ones) and all the U2F browsers (FF+addon, chrome and opera) installed.

Comment (ChristophWurst) [inline templates/challenge.php]: Wow, that is really nice! I'll let you know when there's something testable. You are always welcome to test PRs labed as "3. to review" 🚀

**Pull request merged**